core/templates/pages/community-dashboard-page/services/contribution-opportunities.service.ts

```diff
@@ -17,74 +17,149 @@
  * fields.
  */
 
-require(
-  'pages/community-dashboard-page/services/' +
-  'contribution-opportunities-backend-api.service.ts');
-
-angular.module('oppia').factory('ContributionOpportunitiesService', [
-  'ContributionOpportunitiesBackendApiService',
-  function(ContributionOpportunitiesBackendApiService) {
-    var skillOpportunitiesCursor = null;
-    var translationOpportunitiesCursor = null;
-    var voiceoverOpportunitiesCursor = null;
-    var moreSkillOpportunitiesAvailable = true;
-    var moreTranslationOpportunitiesAvailable = true;
-    var moreVoiceoverOpportunitiesAvailable = true;
-
-    var _getSkillOpportunities = function(cursor, successCallback) {
-      ContributionOpportunitiesBackendApiService.fetchSkillOpportunities(
-        cursor).then(({ opportunities, nextCursor, more }) => {
-        skillOpportunitiesCursor = nextCursor;
-        moreSkillOpportunitiesAvailable = more;
-        successCallback(opportunities, more);
-      });
-    };
-    var _getTranslationOpportunities = function(
-        languageCode, cursor, successCallback) {
-      ContributionOpportunitiesBackendApiService.fetchTranslationOpportunities(
-        languageCode, cursor).then(({ opportunities, nextCursor, more }) => {
-        translationOpportunitiesCursor = nextCursor;
-        moreTranslationOpportunitiesAvailable = more;
-        successCallback(opportunities, more);
-      });
-    };
-    var _getVoiceoverOpportunities = function(
-        languageCode, cursor, successCallback) {
-      ContributionOpportunitiesBackendApiService.fetchVoiceoverOpportunities(
-        languageCode, cursor
-      ).then(function({ opportunities, nextCursor, more }) {
-        voiceoverOpportunitiesCursor = nextCursor;
-        moreVoiceoverOpportunitiesAvailable = more;
-        successCallback(opportunities, more);
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { ContributionOpportunitiesBackendApiService } from
+  // eslint-disable-next-line max-len
+  'pages/community-dashboard-page/services/contribution-opportunities-backend-api.service';
+import { ExplorationOpportunitySummary } from
+  'domain/opportunity/ExplorationOpportunitySummaryObjectFactory';
+import { SkillOpportunity } from
+  'domain/opportunity/SkillOpportunityObjectFactory';
+
+
+interface SkillContributionOpportunities {
+  opportunities: SkillOpportunity[];
+  nextCursor: string;
+  more: boolean;
+}
+
+interface TranslationContributionOpportunities {
+  opportunities: ExplorationOpportunitySummary[];
+  nextCursor: string;
+  more: boolean;
+}
+
+interface VoiceoverContributionOpportunities {
+  opportunities: ExplorationOpportunitySummary[];
+  nextCursor: string;
+  more: boolean;
+}
```

Comment:
import these interfaces from backend-api (filename: contribution-opportunities-backend-api.service). You may have to add export in front of interfaces in that file.